libraries/AP_Scheduler/AP_Scheduler.cpp

```diff
@@ -160,9 +160,9 @@ void AP_Scheduler::run(uint32_t time_available)
     for (uint8_t i=0; i<_num_tasks; i++) {
         const AP_Scheduler::Task& task = (i < _num_unshared_tasks) ? _tasks[i] : _common_tasks[i - _num_unshared_tasks];
 
-        uint32_t dt = _tick_counter - _last_run[i];
+        uint16_t dt = _tick_counter - _last_run[i];
         // we allow 0 to mean loop rate
-        uint32_t interval_ticks = (is_zero(task.rate_hz) ? 1 : _loop_rate_hz / task.rate_hz);
+        uint16_t interval_ticks = (is_zero(task.rate_hz) ? 1 : _loop_rate_hz / task.rate_hz);
```

Comment:
Pretty sure this patch isn't required.  Someone running their loops *much* faster than the slowest task might end up getting problems in here.  e.g. `_loop_rate_hz` of 1000 and a `task.rate_hz` of 1/60 gives an interval_ticks of 60,000 - perilously close the uint16-max.  I think we can afford to burn an extra 2 bytes on the stack here :-)

Reply:
I wasn't sure about this change either; I changed the `uint32_t` to `uint16_t` for sake of consistency with `dt`.

One thing is sure: with the current scheduler architecture, we cannot schedule tasks with an interval larger than 65535 ticks anyway because the tick counter is 16-bit. Therefore,

* If we keep `interval_ticks` as `uint32_t`, it will always be larger than the maximum possible value of `dt` for such tasks if `dt` is 16-bit, so tasks would never get scheduled.
* If we convert `interval_ticks` to `uint16_t`, it could potentially overflow for such rarely-scheduled tasks, and it could potentially overflow in a way that the task gets scheduled much more frequently than expected.

So, it seems like `uint32_t` is safer as we don't cause extra CPU load even if the user misconfigures the loop.